File: server/apps/api-service/src/main/java/com/sharetimer/apiservice/config/TimerConfig.java
package com.sharetimer.apiservice.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.data.redis.core.StringRedisTemplate;
import com.sharetimer.storage.redis.config.RedisTemplateFactory;
import com.sharetimer.storage.redis.config.TimerRedisProps;

@Configuration
public class TimerConfig {

  @Bean("timerExpirationRedisTemplate")
  public StringRedisTemplate timerExpirationRedisTemplate(RedisTemplateFactory redisTemplateFactory,
      TimerRedisProps timerRedisProps) {
    int dbIndex = timerRedisProps.getExpiration().getDbIndex();
    return redisTemplateFactory.getTemplate(dbIndex);
  }

  @Bean("timerPubSubRedisTemplate")
  public StringRedisTemplate timerPubSubRedisTemplate(RedisTemplateFactory redisTemplateFactory,
      TimerRedisProps timerRedisProps) {
    int dbIndex = timerRedisProps.getPubSub().getDbIndex();
    return redisTemplateFactory.getTemplate(dbIndex);
  }

}

File: server/apps/api-service/src/main/kotlin/com/sharetimer/apiservice/config/TimerConfig.kt
package com.sharetimer.apiservice.config

import com.sharetimer.storage.redis.config.RedisTemplateFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.redis.core.StringRedisTemplate
import java.time.Clock

@Configuration
class TimerConfig {
    @Bean
    fun clock(): Clock = Clock.systemUTC()

    @Bean("timerExpirationRedisTemplate")
    fun timerExpirationRedisTemplate(redisTemplateFactory: RedisTemplateFactory): StringRedisTemplate =
        redisTemplateFactory.template

    @Bean("timerPubSubRedisTemplate")
    fun timerPubSubRedisTemplate(redisTemplateFactory: RedisTemplateFactory): StringRedisTemplate =
        redisTemplateFactory.template
}
